designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+compile-time warning notifying the user. This must contain the above 3 criteria:
+1. What is deprecated
+2. How to adjust your code
+3. When you need to adjust it by.
+
+The final bullet point should be decided during the review process with 3
+versions as a minimum. Many changes, such as reserving a new keyword, will not
+require longer periods, but syntax changes which the user may need time to
+adjust to could require longer time frames.
+
+This information must also be added to the documentation in the accompanying PR
+to the change.
+
+Finally, developers may wish to weigh in on how the implementation might look
+after the removal of the deprecated feature as part of the text of the pull
+request, if they can foresee code improvements which can occur at that time.
+
+## Code reviewers
+From a code reviewer standpoint, it is important to ask the following questions:
+1. Is this deprecation tied to a useful feature or noticeable improvement to the
+   underyling code (not just 'for the sake of it')?
+2. Does the new warning meet the criteria above and explain itself clearly?
+3. Is there a simple (preferably automated) way for users to accommodate this
+   change, e.g. through the `print-canonical` flag to stanc3 or a
+   find-and-replace feature of their text editor?
+4. Is the removal timeline appropriate for this change?
+
+## Releases
+Finally, during the release cycle an additional bullet point should be added to
+the release checklist to ensure that the deprecations slated for removal in the
+coming version have been properly handled. To accommodate this, the documentation
+page on deprecations should be sorted/categorized by removal version.
+
+Additionally, the release notes should contain a section on **Deprecations** and
+**Expired Deprecations** (alt. **Removals**) for each version. This should
+duplicate the information on the what and how of the change from the error
+message and documentation.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Drawbacks to this are rather obvious - backwards compatibility is certainly a
+virtue, and any breaking change will lead to some headaches for a portion of the
+user base, no matter how small. In the limit, we certainly want to avoid any
+large enough change that users prefer not to update (a core reason these changes
+should be tied to improvements in the language, not just for their own sake) and
+any changes which require dedicated rewriting or rethinking of models. This
+desire motivated the definition of 'minor change' above and the restriction to
+primarily automatable syntax changes.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This design allows for us to care for the long term health of the Stan language
+and codebase while clearly communicating our intent to users. Small changes can
+be made to make the development of new features easier without "sneaking up on"
+anyone or breaking things willy-nilly.
```

Comment:
"or breaking things willy-nilly" -> ""

We don't need to remind devs not to break things willy nilly!